canonical/canonical.com
# Implement Jinja logic for building the sliding navigation from navigation.yaml

## Done

- Creates a func in Python, `build_navigation`, that takes and 'id' and 'title' and returns the relevant section on the navigation. It works by parsing the navigation.yaml and applying it to a jinja template.
- Creates the Jinja template and logic needed to build the sliding(mobile) navigation
- Minor changes to the formatting of the navigation.yaml so it is easier to understand and process

## QA

*It is a little annoying to QA as the JS that this relies on has not yet been implemented, but do the best you can from the directions below*
- Open the [demo](https://canonical-com-1305.demos.haus/) and use the inspection tool to find the navigation drop-downs. An easy way to do this is to use crtl+f and search `role="menuitem"`, you should see `li` elements with names like 'product', 'solutions', etc. Within these you will find a `<div class="p-navigation__dropdown-content-mobile">`, this is where the navigation generated from this work can be found.
- Check that the nesting and links that exist in the navigation.yaml are present in the HTML markup
- Check the design matches that of the [sliding nav in the vanilla](https://vanillaframework.io/docs/examples/patterns/navigation/sliding-search?theme=light) *note* there are some minor difference that will be introduced in [this pr](https://github.com/canonical/vanilla-framework/pull/5213)


## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-13241
